infrastructure/src/main/java/org/corfudb/infrastructure/log/SegmentSupervisor.java

```diff
@@ -0,0 +1,181 @@
+package org.corfudb.infrastructure.log;
+
+import com.google.common.annotations.VisibleForTesting;
+import lombok.Builder;
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.io.IOUtils;
+
+import javax.annotation.Nullable;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.nio.channels.FileChannel;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+import java.util.ArrayList;
+import java.util.EnumSet;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.stream.Collectors;
+
+import static org.corfudb.infrastructure.utils.Persistence.syncDirectory;
+
+/**
+ * The class manages corfu's file segments
+ */
+@Builder
+@Slf4j
+public class SegmentSupervisor {
+    @Getter
+    private final ConcurrentMap<Path, SegmentHandle> channels = new ConcurrentHashMap<>();
+    private final Set<FileChannel> channelsToSync = ConcurrentHashMap.newKeySet();
+
```

Comment:
If you can keep a dirty flag inside the segment and on write flag it as true, then you can remove channelsToSync. Scanning the channels is supper cheap considering the open segments number is relatively small. Essentially the scan is cheap.

Reply:
I would like to bring any changes in logic in this PR